core/src/CMakeLists.txt

```diff
@@ -100,7 +100,7 @@ KOKKOS_LINK_TPL(kokkoscore PUBLIC ROCM)
 # FIXME: We need a proper solution to figure out whether to enable
 #        libatomic
 IF (KOKKOS_ENABLE_IMPL_DESUL_ATOMICS AND
-    KOKKOS_ENABLE_OPENMPTARGET)
+    (KOKKOS_ENABLE_OPENMPTARGET OR (CMAKE_CXX_COMPILER_ID STREQUAL XLClang)))
```

Comment:
Add a comment ( "XL requires libatomic for 64 bit compare exchange of complex" or so) to capture the description of the PR.
That metadata would not be available from git otherwise.  Only commit message are kept.